File: TRIEU_SON-LE_LOI/BAI3.cpp
#include <iostream>
#include <fstream>
#include <sstream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {

    ifstream inp("BAI3.inp");
    string line;
    getline(inp, line);

    int n, k;
    stringstream ss(line);

    ss >> n;
    ss >> k;
    ss.clear();

    long long a[n], total_price = 0;
    vector<vector<long long>> ranges(k);

    getline(inp, line);
    inp.close();
    ss.str(line);


    for (int i = 0; i < n ; i++)  {
        ss >> a[i];
        ranges[i % k].push_back(a[i]);
    }

    for (auto range: ranges) {
        sort(range.begin(), range.end());
        long long mid = range[range.size() / 2];

        for (long long val: range) {
            total_price += abs(mid - val);
        }
    }

    ofstream out("BAI3.out");
    out << total_price;
    out.close();

    return 0;
}

File: HSG_HAI_DUONG_22_23/contestants/bach_do/bai3.py
with open("bai3.inp","r") as fi:
    n,m=map(int,fi.readline().split())
    s=list(map(int,fi.readline().split()))
    d=0
    for i in range (m):
        if s[0]>=s[len(s)-1]:
            d+=s[0]
            s.remove(s[0])
        elif s[0]<=s[len(s)-1]:
            d+=s[len(s)-1]
            s.remove(s[len(s)-1])
with open("bai3.out","w") as fo:
    fo.write(str(d))
fi.close()
fo.close()